### PR TITLE
Backend code refactoring - part 1 

### PR DESCRIPTION
**Pull request by a60814billy**
- [x] Extract Auth Method from app.js
- [x] Separate routing method from app.js
- [x] Refactor config.js improve readability

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -154,63 +141,19 @@ server.on('resumeSession', function (id, cb) {
...
-app.use(function (req, res, next) {
-  try {
-    decodeURIComponent(req.path)
-  } catch (err) {
-    logger.error(err)
-    return response.errorBadRequest(res)
-  }
-  next()
-})
+app.use(require('./lib/web/middleware/checkURiValid'))
```
Comment (Yukaii) [inline app.js]: Would it be better if we use the camel case convention `checkUrlValid` ? 😕

Comment (a60814billy) [inline app.js]: The middleware is checking uri only, I'll rename to checkURIValid

**Comment by SISheogorath** [comment]
Very nice! I like the new structure!

**Comment by jackycute** [comment]
Thanks @a60814billy

**Pull request merged**